### ISSUE TITLE
Loading messes up TabbedPage on windows phone

### ISSUE THREAD

**Issue by PaulAchess** [aritchie/userdialogs#32]
Hi !
I experienced a little trouble while creating my windows phone project (silverlight, version 8 or 8.1 produce the bug).

**Background**:
I created a tabbed page (inside a navigationpage) that has 2 tabs, each one containing a list view.
I am using an async method to populate these views.

**Problem**:
If I use a loading dialog to wrap the « populate » method, both pages are displayed together when I switch tabs (on my project I don't even have to change tab). This doesn't happen when I remove it.

**What I was able to do**:
I went through the code but I was unfortunately not good enough to understand how to fix and identify the problem *but* I was able to reproduce on a very small project.

Here you can find the minimum reproduction project:
https://drive.google.com/file/d/0B6kaeItD7VQ_cGRjNXBLMUlBNFk/view?usp=sharing

And here is what it looks like:
![multipletabviews](https://cloud.githubusercontent.com/assets/6310023/8060835/bbc533fc-0ec7-11e5-8dc2-23611027911d.png)

I'm truly sorry as these kind of bugs are really annoying to debug, especially because it might come from the WP toolkit. I hope the reproduction project could help anyone willing to investigate.

I just removed the loading dialog to prevent this from happening.

Thanks a lot for your awesome library !

**Comment by aritchie** [comment]
I can only give you a few pointers to what may help you fix this issue.  

First off, you are calling for the loading in the constructor which is usually not a good idea since the view hasn't really loaded yet.  Move the loading code to onappearing.

Secondly (this example doesn't show it), but make sure you don't have both tabs calling for loading.  If both tabs are calling for a loading window (if you are doing this in the constructor you are technically doing this) at the same time, it will likely produce some weird issues.

**Comment by PaulAchess** [comment]
Hi !

Thanks for your feedback. I was indeed trying to load on the constructor of each page previously, which is why I migrated it to the main TabbedPage. Still, this was in the constructor and I this was  not completely satisfying. Thanks for the OnAppearing lead, I think this is way more proper to do so.

**Migrating the code to OnAppearing of the main page actually fixed the issue**. What was weird is that it only affects WP, while iOS and Android have no problem with that. Also, due to the use of messenger, I did not see I was loading before the page had appeared.

Thank you @aritchie ! I think we can close this issue, unless you want to add something.

If anyone experience this later:
**TL;DR: don't use dialogs in constructors**.
